### ISSUE TITLE
prov/opx: auto enabled, but does not find rdma/hfi/hfi1_user.h

### ISSUE THREAD

**Issue by bsbernd** [ofiwg/libfabric#7842]
...
00:43:59 configure: *** Configuring opx provider
00:43:59 checking for opx provider... checking uuid/uuid.h usability... yes
00:43:59 checking uuid/uuid.h presence... yes
00:43:59 checking for uuid/uuid.h... yes
...
00:44:04 *** Built-in providers:	opx dmabuf_peer_mem hook_hmem hook_debug perf rstream shm tcp udp verbs sockets psm3 
...
00:44:10   CC       prov/opx/src/src_libfabric_la-fi_opx_mr.lo
00:44:10   CC       prov/opx/src/src_libfabric_la-fi_opx_msg.lo
00:44:10 In file included from ./prov/opx/include/opa_service.h:65:0,
00:44:10                  from prov/opx/src/fi_opx_init.c:40:
00:44:10 ./prov/opx/include/opa_common_gen1.h:61:10: fatal error: rdma/hfi/hfi1_user.h: No such file or directory
00:44:10  #include <rdma/hfi/hfi1_user.h>
00:44:10           ^~~~~~~~~~~~~~~~~~~~~~
00:44:10 compilation terminated.
00:44:10 make[1]: *** [prov/opx/src/src_libfabric_la-fi_opx_init.lo] Error 1
00:44:10 make[1]: *** Waiting for unfinished jobs....
00:44:18 make[1]: Leaving directory `/var/lib/jenkins/workspace/IME_gerrit_review/IME_Gerrit_Review_Distributed_RPC_Multi_Servers_test/contrib/nongpl/net/ofi/libfabric'

I guess checking for uuid/uuid.h is not a good test...

```
       FI_CHECK_PACKAGE([opx_uuid],
            [uuid/uuid.h],
            [uuid],
            [uuid_parse],
            [],
            [],
            [],
            [opx_happy=1],
            [opx_happy=0])
```

**Comment by timothom64** [comment]
Looks like the error isn't uuid, it's that you don't have the kernel headers installed?
<rdma/hfi/hfi1_user.h>

That file is needed for various structures and to make sys calls into the device driver.

Reguardless, this need to fail more gracefully if the file cannot be found.  Please assign this bug to me

**Comment by timothom64** [comment]
Fixed with a73601635ad66b89fae0fe953282a63a072b19e0

**Comment by timothom64** [comment]
I think this can be closed?  @shefty ?
